### DISCUSSION TITLE
Losing more hair -- dev builds and osx 10.14 - linker error!

### DISCUSSION THREAD

**Discussion by rlevine** [inkstitch/inkstitch#2063] [General]
TL;DR Can the dev builds include a legacy OSX build, or if that's too painful or it's hand-built, what's the recipe, so I can create them for myself?

***

I'm hitting a fail trying to get inkstitch dev builds to run on osx 10.14, Mojave. (Yes, I have to run on Mojave, because I'm working with machinery that's as old as snot, but that's a longer story.) I've tracked down all the red herrings the error leads to (text attached below) 
and after lots of futile poking at gettext and libintl.8.dylib versions, the error turns out to be that Apple changed the linker incompatibly between 10.14 and 10.15, and the linker in newer osx sdks spits out load commands for dylibs that aren't supported in 10.14. It looks like Xcode 13.2 will work, or whatever is being used in the current legacy build is fine.

I haven't been able to spot the build mechanics for the legacy osx build in the tree; apologies if it's obvious, but I can't see it. :(

***

```
[28779] Error loading Python lib '/Users/CS255/Library/Application Support/org.inkscape.Inkscape/config/inkscape/extensions/inkstitch/inkstitch 2.app/Contents/MacOS/libpython3.8.dylib': dlopen: dlopen(/Users/CS255/Library/Application Support/org.inkscape.Inkscape/config/inkscape/extensions/inkstitch/inkstitch 2.app/Contents/MacOS/libpython3.8.dylib, 10): Library not loaded: @loader_path/libintl.8.dylib
  Referenced from: /Users/CS255/Library/Application Support/org.inkscape.Inkscape/config/inkscape/extensions/inkstitch/inkstitch 2.app/Contents/MacOS/libpython3.8.dylib
  Reason: no suitable image found.  Did find:
	/Users/CS255/Library/Application Support/org.inkscape.Inkscape/config/inkscape/extensions/inkstitch/inkstitch 2.app/Contents/MacOS/libintl.8.dylib: cannot load 'libintl.8.dylib' (load command 0x80000034 is unknown)
	/Users/CS255/Library/Application Support/org.inkscape.Inkscape/config/inkscape/extensions/inkstitch/inkstitch 2.app/Contents/MacOS/libintl.8.dylib: stat() failed with errno=3
	/Users/CS255/Library/Application Support/org.inkscape.Inkscape/config/inkscape/extensions/inkstitch/inkstitch 2.app/Contents/MacOS/libintl.8.dylib: cannot load 'libintl.8.dylib' (load command 0x80000034 is unknown)
	/Users/CS255/Library/Application Support/org.inkscape.Inkscape/config/inkscape/extensions/inkstitch/inkstitch 2.app/Contents/MacOS/libintl.8.dylib: stat() failed with errno=3
```

**Comment by lexelby** [comment]
I think @kaalleen or @rejbasket builds the legacy versions manually on their computer and uploads then when we do releases.  It's kinda terrible but there's no way we've been and to find to use an older xcode on GitHub actions.

For awhile I toyed with the idea of setting up a github build agent an old MacBook Air I have kicking around, but the last time I looked, there were some pretty scary security issues with that.

**Reply by rlevine** [reply]
Hmm. Breadcrumbs?  https://github.com/devernay/xcodelegacy

I’m happy to build my own, if I could get a copy of the build files.

**Reply by lexelby** [reply]
All of our build stuff is in .github/, which uses the Makefile and a few scripts in bin/.

**Comment by kaalleen** [comment]
Have a look over here: https://inkstitch.org/developers/inkstitch/manual-build-old-macos/

**Comment by rlevine** [comment]
Thanks! I did it backwards, manual install first and then figured out what make wanted. I'm using conda, as it versions the lib stack. (And I'm a wimp, and trip over my own feet when I mess up the pyenv path shims. :) 

Breadcrumbs for anyone else who needs to build on a Mojave-ish box, assuming you have a version-appropriate Xcode installed. This works under 11.3.1:

\# Create a conda python 3.8 env assuming miniconda is installed: 

`conda create --name inkstitch python=3.8`
`conda activate inkstitch`

\# and follow the Ink/Stitch [manual install instructions](https://inkstitch.org/developers/inkstitch/manual-setup/). Ignore creating the symbolic link in step 6, as the goal of this is to create an installable package. (And more recent OSXes put inkscape extensions in `/Users/CS255/Library/Application\ Support/org.inkscape.Inkscape/config/inkscape/extensions`.)

\# `pyinstaller` isn't included in `requirements.txt`
`pip install pyinstaller`

\# In the shell:
`export BUILD=osx`

 And if you get a strange node.js issue with SSL, depending on the node version you have installed:
`export NODE_OPTIONS=--openssl-legacy-provider`

\# `VERSION` gets stepped on by the first line of `bin/build-distrubution-archives` so add this second line.
\# It'll name the build package file with your git branch name, assuming a format like `<your_name>/<feature_name>`.
`VERSION='$(command -v git) branch --show-current | sed 's/[\/]/-/g'`


\# and... 

`make` 

\# ...in the top-level directory, needing one `y` response to let it overwrite `dist/inkstitch.app`.

\# An installable package will be dropped in `artifacts`.
